src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
+        for (int i = 0; i < 6; i++) {
+            if (i < 3) {

```

Comment:
if you wanted to change array size, you would need to change 3 values: 6, 6, 3. These values are logically connected, and good code should reflect that: create a variable length, and use length/2 as an `if` condition